src/vmm/src/persist.rs

```diff
@@ -392,13 +392,30 @@ pub fn restore_from_snapshot(
     vm_resources: &mut VmResources,
 ) -> Result<Arc<Mutex<Vmm>>, RestoreFromSnapshotError> {
     let microvm_state = snapshot_state_from_file(&params.snapshot_path)?;
+    let track_dirty_pages = params.enable_diff_snapshots;
+
+    let vcpu_count = microvm_state
+        .vcpu_states
+        .len()
+        .try_into()
+        .map_err(|_| VmConfigError::InvalidVcpuCount)
```

Comment:
Is it fine to use `InvalidVcpuCount` error which indicates "The number of vCPUs must be greater than 0, less than {MAX_SUPPORTED_VCPUS:} and must be 1 or an even number if SMT is enabled." for conversion from `usize` to `u8`? Create a new error variance for it or update the error message to cover it instead?

Reply:
I think this is fine, because if the conversion fails, then the vcpu count was > 255, which falls under "must be [...] less than MAX_SUPPORTED_VCPUS"

Reply:
ah you're right. please forget it.